primitives/core/src/traits.rs

```diff
@@ -163,15 +164,28 @@ pub trait BareCryptoStore: Send + Sync {
 	/// each key given that the key is supported.
 	///
 	/// Returns a list of `Result`s each representing the SCALE encoded
-	/// signature of each key or a BareCryptoStoreError for non-supported keys.
+	/// signature of each key or a Error for non-supported keys.
 	fn sign_with_all(
 		&self,
 		id: KeyTypeId,
 		keys: Vec<CryptoTypePublicPair>,
 		msg: &[u8],
-	) -> Result<Vec<Result<Vec<u8>, BareCryptoStoreError>>, ()>{
+	) -> Result<Vec<Result<Vec<u8>, Error>>, ()>{
 		Ok(keys.iter().map(|k| self.sign_with(id, k, msg)).collect())
 	}
+
+	/// Generate VRF proof for given transacript data.
```

Comment:
These docs have a lot of room for improvement. It does not specify any assumptions of the caller, or anything about the return type. Under what conditions it should be `Ok`, `Err`, `Some`, `None`, or even what the `(Vec<u8>, Vec<u8>)` tuple is. From reading this function declaration and documentation I have absolutely no idea what it is supposed to do. And I know what a VRF is and how ristretto instantiates it.

Reply:
I hope the docs are not much better.